lib/ansible/modules/network/asa/asa_og.py

```diff
@@ -0,0 +1,804 @@
+            if group_object:
+                for i in group_object:
+                    if i in have_group_object:
+                        if 'object-group network {0}'.format(name) not in commands:
+                            commands.append('object-group network {0}'.format(name))
+                        commands.append('no group-object ' + i)
+            if address:
+                for i in address:
+                    if i in have_ip_mask:
+                        if 'object-group network {0}'.format(name) not in commands:
+                            commands.append('object-group network {0}'.format(name))
+                        commands.append('no network-object ' + i)
+
+    elif 'port-object' in group_type:
+
+        if have_group_type is None and have_protocol is None:
+            return commands
+
+        elif 'port' in have_group_type and have_protocol == protocol:
+
+            if port_range:
+                for i in port_range:
+                    if i in have_port_range:
+                        if 'object-group service {0} {1}'.format(name, protocol) not in commands:
+                            commands.append('object-group service {0} {1}'.format(name, protocol))
+                        commands.append('no port-object range ' + i)
+            if port_eq:
+                for i in port_eq:
+                    if i in have_port_eq:
+                        if 'object-group service {0} {1}'.format(name, protocol) not in commands:
+                            commands.append('object-group service {0} {1}'.format(name, protocol))
+                        commands.append('no port-object eq ' + i)
+            if description:
+                if description == have_description:
+                    if 'object-group service {0} {1}'.format(name, protocol) not in commands:
+                        commands.append('object-group service {0} {1}'.format(name, protocol))
+                    commands.append('no description {0}'.format(description))
+
+    elif 'service-object' in group_type:
+
+        if have_group_type is None:
+            return commands
+
+        elif 'service' in have_group_type:
+            if description:
+                if description == have_description:
+                    if 'object-group service {0}'.format(name) not in commands:
+                        commands.append('object-group service {0}'.format(name))
+                    commands.append('no description {0}'.format(description))
+            if service_cfg:
+                for i in service_cfg:
+                    if i in have_service_cfg:
+                        if 'object-group service {0}'.format(name) not in commands:
+                            commands.append('object-group service {0}'.format(name))
+                        commands.append('no service ' + i)
+
+    return commands
+
+
+def map_obj_to_commands(want, have, module):
+
+    global have_name
```

Comment:
We discourage using `global`. Please pass the variables to the required functions.